api/v1alpha1/azurevirtualmachineextension_types.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// AzureVirtualMachineExtensionSpec defines the desired state of AzureVirtualMachineExtension
+type AzureVirtualMachineExtensionSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	Location                string `json:"location"`
+	ResourceGroup           string `json:"resourceGroup"`
+	VMName                  string `json:"vmName"`
+	AutoUpgradeMinorVersion bool   `json:"autoUpgradeMinorVersion"`
+	ForceUpdateTag          string `json:"forceUpdateTag"`
+	Publisher               string `json:"publisher"`
+	TypeName                string `json:"typeName"`
+	TypeHandlerVersion      string `json:"typeHandlerVersion"`
+	Settings                string `json:"settings,omitempty"`
+	ProtectedSettings       string `json:"protectedSettings,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// AzureVirtualMachineExtension is the Schema for the azurevirtualmachineextensions API
+// +kubebuilder:resource:shortName=vmext,path=azurevirtualmachineextension
```

Comment:
```suggestion
// +kubebuilder:resource:shortName=vmext,path=azurevirtualmachineextensions
```

needs to be plural ...if it works for you it might be a kustomize version mismatch